make giter command to run on terminal without configuration

vccolombo opened the pull request:
Hello @Xcal1bur! After reading your reply in #1 I thought that all this configuration necessary to run the giter command was easy to get rid, so I decided to make this PR and I would love to know what you think about it.

Before this PR, one would need to download this repo and configure a lot of things to make giter command to run on terminal. With simple additions to setup.py, this PR creates an entry point that makes the command giter available out-of-the-box after running pip install .

How to use now:
```terminal
pip install . # in giter root folder, to install giter
giter # anywhere, just to check if the terminal command is working
```

giter should still work in the old way (like in #1) after this PR

Commit:
make giter command to run on terminal without configuration

Before this commit, one would need to download this repo and configure a lot of things to make giter command to run on terminal. With simple additions to setup.py, this commit creates an entry points that makes the command giter available out-of-the-box after running pip install .

Xcal1bur commented:
Hey @vccolombo! Thanks for putting your time and work in this PR. I really appreciate it!

I really like your PR and would like to merge it. I tried to install as you proposed by cloning your fork and running `sudo pip install .` in the root directory. It installed successfully though I noticed that it was installed to `/usr/lib/python3.7/site-packages/` which is included in `PYTHONPATH` but not in the shell `$PATH` meaning that the command will not be found when trying to execute `giter` in the terminal. This either means one would need to add `/usr/lib/python3.7/site-packages/` to the `$PATH` which be the same issue you proposed to fix in this PR or we need to find a way to install giter or a hard-/symbolic link to a directory listed in the `$PATH` like `usr/local/bin/`.

Xcal1bur commented:
Hey @vccolombo! Thank you very much for putting your time and work in this PR. I really appreciate it!

I really like your solution! Just one small thing I'd change. `src/` should be renamed to `giter/` since pip otherwise installs it in `/usr/lib/python3.7/site-packages/src/` as opposed to `/usr/lib/python3.7/site-packages/giter/`.

Commit:
Renamed src/ to giter/

Commit:
Changed imports and entry_point to account for changed directory name

Commit:
Changed installation instructions

Xcal1bur commented:
@vccolombo do you have something to add or did forget something? Ready for merge?

Commit:
Change git clone link

The old link was an SSH link, that could cause problems to users that did not configure SSH for their account

vccolombo commented:
Hey! Thanks for reviewing my PR and doing the changes!

I just have one question about the README changes. Is it necessary to run `sudo pip install .`? Why does it need to be run with sudo? 

This is the only issue I would have, the code itself is ready to merge

Xcal1bur commented:
Well because running it without sudo raises an error for me:
`Could not install packages due to an EnvironmentError: [Errno 13] Permission denied: '/usr/bin/giter'
Consider using the --user option or check the permissions.`. 
Non-root users do not have write access to `/usr/bin/`.

vccolombo commented:
I think a better idea would be to install it using `pip install --user .`, have you tried that?

Xcal1bur commented:
`The script giter is installed in '/home/xcalibur/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.`
So you would need to this to the `$PATH` as well. Is it working for you without that warning?

vccolombo commented:
I don't receive this warning. That is weird, I don't think this warning should happen. Are you able to install other pip packages without sudo?

Just for some clarification about why you shouldn't install pip packages with sudo:

https://askubuntu.com/questions/802544/is-sudo-pip-install-still-a-broken-practice
https://stackoverflow.com/questions/15028648/is-it-acceptable-and-safe-to-run-pip-install-under-sudo
https://dev.to/elabftw/stop-using-sudo-pip-install-52mn

Xcal1bur commented:
Thanks for the links. I will refrain from using `sudo pip ...` in the future.
I'm able to install other packages without warnings. I'll figure it out. As long as it is working for everybody else without issues it's fine I guess. I'll change the README.

Commit:
Changed installation instructions

Xcal1bur commented:
Thanks a lot for contributing! I really appreciate it!
